cmd/kubeadm/app/phases/upgrade/versiongetter.go

```diff
@@ -122,3 +122,30 @@ func computeKubeletVersions(nodes []v1.Node) map[string]uint16 {
 	}
 	return kubeletVersions
 }
+
+// OfflineVersionGetter will use the version provided or
+type OfflineVersionGetter struct {
+	VersionGetter
+	version string
+}
+
+// NewOfflineVersionGetter wraps a VersionGetter and skips online communication if default information is supplied.
+// Version can be "" and the behavior will be identical to the versionGetter passed in.
+func NewOfflineVersionGetter(versionGetter VersionGetter, version string) VersionGetter {
```

Comment:
The creation of a different struct kind of makes me bristle, vs. falling back within the existing one, but it's not worth churning over.

Reply:
That was a little more invasive, but I know what you mean.